Update BitGoJS

jayd3e opened the pull request:
Updates BitGoJS to the latest version.

CapnMigraine committed:
TOKS-31 - Allow sending trustline txs from SDK

Commit:
SDKT-32: Improve usability for typescript dependents

* Add @types/bluebird and @types/superagent to the deps list from dev deps,
  since these types are exported by the BitGoJS public API
* Export BitGo object interfaces
* Return custom `BitGoRequest` instead of raw superagent Request.
  Methods patched onto the superagent Request object will be declared on
  this interface instead of augmenting superagent.Request.
* Separate source files from type declarations in dist/

This commit covers the following tickets:
* SDKT-32
* SDKT-33
* SDKT-31

Commit:
SDKT-26: Fix SDK Documentation link in README, add README to example/

perrybitgo committed:
chore(release): publish modules

 - bitgo@8.2.2
 - @bitgo/express@8.2.2

perrybitgo committed:
Update SDK to bitgo@^8.2.2

Commit:
Merge pull request #541 from BitGo/SDKT-26-fix-readme-links

SDKT-26: Fix SDK Documentation link in README, add README to example/

Commit:
Merge pull request #542 from BitGo/rel/innsbruck

Rel/innsbruck

Commit:
Merge pull request #536 from BitGo/TOKS-31.send-change-account-tx

TOKS-31 - Allow sending trustline txs from SDK

Commit:
SDKT-24: Add missing options types in wallet and wallets

Commit:
SDKT-4: Enable more strict tsc options

Commit:
SERV-58: Fix environment inheritance ordering

Commit:
SERV-263: Move statics into package for import into BitGoJS monorepo

Commit:
SDKT-4: Clean up formatted token config initialization

Commit:
Merge pull request #527 from BitGo/SDKT-3-typescript-express-clientroutes

SDKT-3: Improve typescript support and usage in express clientRoutes

Commit:
Merge pull request #540 from BitGo/SDKT-32-types-deps

SDKT-32: Improve usability for typescript dependents

Commit:
Merge pull request #545 from BitGo/SERV-58-fix-environment-inheritance

SERV-58: Fix environment inheritance ordering

Commit:
SDKT-4: Fix post-rebase errors

CapnMigraine committed:
TOKS-32 - Add domain property to Stellar tokens

Commit:
SERV-269 update express docker production docs

Commit:
Merge pull request #547 from BitGo/SERV-269-update-express-prod-docs

SERV-269 update express docker production docs

twtaylorbitgo committed:
TRON-3 - hot wallet sending

CapnMigraine committed:
TOKS-33 Allow removing XLM trustlines

Commit:
Merge pull request #63 from BitGo/TOKS-32.add-domain-stellar-tokens

TOKS-32 - Add domain property to Stellar tokens

Commit:
Merge pull request #549 from BitGo/TOKS-33.allow-removing-xlm-trustlines

TOKS-33 Allow removing Stellar trustlines

OttoAllmendinger committed:
Remove params.targetWalletUnspents

This is a phantom parameter that was described for the `consolidate`
route (but does not exist there either).

Issue: BG-14108

OttoAllmendinger committed:
v2/wallets.ts: combine consolidate/fanout

* Extract common ports of both funcs into `manageUnspents`
* Fix documentation
* Use `this` instead of `self` (with `.bind(this)`)

Issue: BG-14108

OttoAllmendinger committed:
wallet.js: consolidate: remove prevId

This parameter also does not actually exist.

Issue: BG-14108

Commit:
Merge pull request #544 from BitGo/SDKT-4-tsc-options

SDKT-4: Enable more strict tsc options

OttoAllmendinger committed:
wallets.ts: mark `manageUnspents` private

Also use `self = this` again.

Issue: BG-14108

Commit:
Merge pull request #551 from BitGo/BG-14108.add-manageunspents

Add new method `wallet.manageUnspents`

CapnMigraine committed:
TOKS-38 - Add BitGo token on Stellar mainnet

CapnMigraine committed:
Merge pull request #64 from BitGo/TOKS-38.add-bitgo-stellar-mainnet-token

TOKS-38 - Add BitGo token on Stellar mainnet

argjv committed:
Fix request params

CapnMigraine committed:
TOKS-55 - Add more Stellar tokens to statics config

argjv committed:
Add signature check

argjv committed:
Update yarn.lock

argjv committed:
Merge pull request #548 from BitGo/TRON-3-send-trx

TRON-3 - hot wallet sending

Commit:
Merge pull request #543 from BitGo/SDKT-24-typescript-wallet

SDKT-24: Add missing options types in wallet and wallets

Commit:
Added new erc20 tokens

barathcj committed:
Added underlying asset for yng

barathcj committed:
Changed yng address

barathcj committed:
remove debug log

CapnMigraine committed:
Merge pull request #66 from BitGo/TOKS-55.add-more-stellar-tokens

TOKS-55 - Add more Stellar tokens to statics config

barathcj committed:
Merge pull request #67 from BitGo/BLOCK-92-changed-ERC20-Tokens

Added new erc20 tokens

perrybitgo committed:
chore(release): publish modules

 - bitgo@8.4.0-rc.0
 - @bitgo/express@8.4.0-rc.0

perrybitgo committed:
chore(release): publish modules

 - bitgo@8.4.0-rc.0
 - @bitgo/express@8.4.0-rc.0

perrybitgo committed:
chore(release): publish modules

 - bitgo@8.4.0-rc.0
 - @bitgo/express@8.4.0-rc.0

glowang committed:
TRON-43: Explain a Tron transaction from txHex
* Deserialize a txHex to extract information to display to customer
  before a tx is sent

perrybitgo committed:
DO-387. Update package.json

perrybitgo committed:
Merge pull request #68 from BitGo/DO-387-UpdateStaticsVersion

DO-387. Update package.json

perrybitgo committed:
DO-387. Update package.json rc0

perrybitgo committed:
Merge pull request #69 from BitGo/DO-387-UpdateStaticsVersion-rc0

DO-387. Update package.json 2.3.0-rc0

Commit:
SERV-443: Upgrade @bitgo/statics to 2.3.0-rc.0

Commit:
SERV-443: Fix hardcoded unit test

Commit:
Merge pull request #555 from BitGo/SERV-443-upgrade-statics

SERV-443: Upgrade @bitgo/statics to 2.3.0-rc.0

james8nguyen committed:
DO-395 Rebased from master @ 05d470e780ef679fa8779aa4143d562d069e686a to rel/kansas to include fixes.

CapnMigraine committed:
TOKS-66 - Use base units in Stellar trustlines

Commit:
SERV-445: Update https-proxy-agent to ^3.0.0

james8nguyen committed:
chore(release): publish modules

 - bitgo@8.4.0-rc.1
 - @bitgo/express@8.4.0-rc.1

james8nguyen committed:
chore(release): publish modules

 - bitgo@8.4.0-rc.1
 - @bitgo/express@8.4.0-rc.1

CapnMigraine committed:
TOKS-70 - Add Testnet Stellar token for testing

perrybitgo committed:
DO-413: Use third digit as rc build number

james8nguyen committed:
DO-413: Merge pull request #71 from BitGo/do-413-remove-rc

DO-413: Use third digit as rc build number

Commit:
Merge pull request #560 from BitGo/SERV-445-update-https-proxy-agent

SERV-445: Update https-proxy-agent to ^3.0.0

argjv committed:
Change signTransaction return type

Commit:
Merge pull request #70 from BitGo/TOKS-70.add-testnet-stellar-token

TOKS-70 - Add Testnet Stellar token for testing

perrybitgo committed:
DO-412: Do not hard code statics version.

perrybitgo committed:
Merge pull request #563 from BitGo/do-412-updateStaticsVersion

DO-412: Do not hard code statics version.

perrybitgo committed:
Merge branch 'rel/kansas' of github.com:BitGo/BitGoJS into rel/kansas

perrybitgo committed:
chore(release): publish modules

 - bitgo@8.4.0-rc.2
 - @bitgo/express@8.4.0-rc.2

argjv committed:
Fix review comments

argjv committed:
Merge pull request #564 from BitGo/TRON-76-fix-pending-approvals

TRON-76: Change signTransaction return type

Commit:
Merge remote-tracking branch 'statics/SERV-263-import-statics' into SERV-263-import-statics

Commit:
SERV-263: Update yarn.lock for statics import

Commit:
SERV-263: Clean up loose ends from statics import

* Merge CODEOWNERS files
* Remove package-lock.json
* Update repo url in package.json
* Extend root tsconfig, remove unneeded overrides

Commit:
SERV-263: Continue using node 10 for integration tests for now

Commit:
SERV-456: Use node 10 instead of LTS in drone pipelines, for now

Commit:
SERV-456: Use node 10 for Express docker base container

twtaylorbitgo committed:
Merge pull request #565 from BitGo/SERV-456-node-10

SERV-456: Use node 10 instead of LTS

Commit:
Merge pull request #546 from BitGo/SERV-263-import-statics

SERV-263: Import statics

Commit:
SERV-263: Update README for new statics module

glowang committed:
TRON-43: use accountLib function for tx signing
 *git rid of all references to tron lib and substitute in
 account lib functions instead

Commit:
SERV-460: Add envs for ms domains

Commit:
SERV-460: Add envs for ms domains

Commit:
Merge pull request #567 from BitGo/SERV-263-update-readme-statics

SERV-263: Update README for new statics module

glowang committed:
TRON-43: Update accountlib import

glowang committed:
Merge pull request #554 from BitGo/Tron-43-explain-tx

TRON-43: Explain a Tron transaction from txHex

dannydeezy committed:
SERV-460: Merge pull request #568 from BitGo/SERV-460-ms-envs

SERV-460: Add envs for ms domains

dannydeezy committed:
Merge pull request #568 from BitGo/SERV-460-ms-envs

SERV-460: Add envs for ms domains

james8nguyen committed:
chore(release): publish modules

 - bitgo@8.4.0
 - @bitgo/express@8.4.0

CapnMigraine committed:
Merge pull request #557 from BitGo/TOKS-66.stellar-trustline-use-base-units

TOKS-66 - Use base units in Stellar trustlines

Commit:
[BLOCK-126] Add ERC20 Tokens

AMO Token (AMO) -- 0x38c87aa89b2b8cd9b95b736e1fa7b612ea972169
Ecotech Visions (ETV) -- 0x695F5d0692655ebf710c0003ca499323488334C6
MFPH Token (MFPH) -- 0xbc7e1056ecc72d14228a14a53815c5d3ad86c84e

Commit:
Merge branch 'master' into merge-kansas

perrybitgo committed:
Merge pull request #571 from BitGo/merge-kansas

Merge rel/kansas

CapnMigraine committed:
TOKS-79 - Add suffix to Stellar tokens in statics

marktoda committed:
Merge pull request #569 from BitGo/BLOCK-126-add-ERC20-Tokens

[BLOCK-126] Add ERC20 Tokens

perrybitgo committed:
Update release version

Commit:
Merge branch 'master' into merge-kansas2

perrybitgo committed:
Merge pull request #574 from BitGo/merge-kansas2

Merge kansas pt 2

CapnMigraine committed:
Merge pull request #572 from BitGo/TOKS-79.add-stellar-token-suffix

TOKS-79 - Add suffix to Stellar tokens in statics

argjv committed:
Make outputs field an array

Fix unit tests

Fix fee format

Fix review comments

argjv committed:
Merge pull request #575 from BitGo/TRON-60-fix-explain-tx-format

TRON-60: Make outputs field an array

argjv committed:
Update Account lib and fix tests

argjv committed:
Merge pull request #576 from BitGo/TRON-95-update-account-lib

TRON-95: Update Account lib and fix tests

argjv committed:
Allow trx to sign with an xprv

argjv committed:
Merge pull request #578 from BitGo/TRON-100-sign-with-xprv

Allow trx to sign with an xprv

Commit:
ACID-69: Change ofcxlm to have correct decimalPlaces

Commit:
SERV-484: Recreate XLM integration test wallets

The XLM testnet was reset earlier today, and the previous integration
test wallets have been deleted.

This commit sets the wallet ids to newly created XLM wallets, as well as
some minor cleanup of duplicated test code and missing types.

Commit:
BLOCK-108 allow unspent reservation param in build

Commit:
Merge pull request #580 from BitGo/SERV-484-xlm-integration-test-wallets

SERV-484: Recreate XLM integration test wallets

Commit:
Merge pull request #579 from BitGo/ACID-69-ofcxlm-fix

ACID-69: Change ofcxlm to have correct decimalPlaces

dannydeezy committed:
Merge pull request #581 from BitGo/BLOCK-108-allow-unspent-reservation-in-build

BLOCK-108 allow unspent reservation param in build

glowang committed:
TRON-103: integrate 0.1.3 ver of account lib in SDK

glowang committed:
Merge pull request #582 from BitGo/TRON-103-update-account-lib-reference

TRON-103: integrate 0.1.3 ver of account lib in SDK

perrybitgo committed:
DO-494: Update statics to 2.4.0-rc, bitgo-account-lib to 0.1.4 in BitGoJS

perrybitgo committed:
Merge pull request #583 from BitGo/do-494-update-statics-account-lib

DO-494: Update statics to 2.4.0-rc, bitgo-account-lib to 0.1.4 in Bit…

perrybitgo committed:
chore(release): publish modules

 - bitgo@8.5.0-rc.1
 - @bitgo/express@8.5.0-rc.1
 - @bitgo/statics@2.4.0-rc.1

perrybitgo committed:
chore(release): publish modules

 - bitgo@8.5.0-rc.1
 - @bitgo/express@8.5.0-rc.1
 - @bitgo/statics@2.4.0-rc.1

Commit:
Remove TRX Derive Key With Seed

The TRX implementation was throwing "not implemented" error even though
the base coin had a perfectly good implementation that works fine for
TRON. Remove the override implementation in the tron coin

Commit:
Remove TRX Derive Key With Seed

The TRX implementation was throwing "not implemented" error even though
the base coin had a perfectly good implementation that works fine for
TRON. Remove the override implementation in the tron coin

twtaylorbitgo committed:
Merge pull request #584 from BitGo/TRON-107-add-trx-deriveKeyWithSeed

Remove TRX Derive Key With Seed

twtaylorbitgo committed:
Merge pull request #584 from BitGo/TRON-107-add-trx-deriveKeyWithSeed

Remove TRX Derive Key With Seed

twtaylorbitgo committed:
TRON-109 - valid address need to be over base58 and hex address implementation

twtaylorbitgo committed:
TRON-109 - valid address need to be over base58 and hex address implementation

twtaylorbitgo committed:
- Prettier linting change

twtaylorbitgo committed:
- Prettier linting change

marktoda committed:
Merge pull request #586 from BitGo/TRON-109-address-in-tron-broken

TRON-109 - valid address need to be over base58 and hex addresses

marktoda committed:
Merge pull request #586 from BitGo/TRON-109-address-in-tron-broken

TRON-109 - valid address need to be over base58 and hex addresses

perrybitgo committed:
Update dependencies

perrybitgo committed:
Update dependencies

perrybitgo committed:
chore(release): publish modules

 - bitgo@8.5.0-rc.2
 - @bitgo/express@8.5.0-rc.2

perrybitgo committed:
chore(release): publish modules

 - bitgo@8.5.0-rc.2
 - @bitgo/express@8.5.0-rc.2

perrybitgo committed:
chore(release): publish modules

 - bitgo@8.5.0-rc.3
 - @bitgo/express@8.5.0-rc.3

perrybitgo committed:
chore(release): publish modules

 - bitgo@8.5.0-rc.3
 - @bitgo/express@8.5.0-rc.3

perrybitgo committed:
chore(release): publish modules

 - bitgo@8.5.0
 - @bitgo/express@8.5.0
 - @bitgo/statics@2.4.0

perrybitgo committed:
chore(release): publish modules

 - bitgo@8.5.0
 - @bitgo/express@8.5.0
 - @bitgo/statics@2.4.0

Commit:
SERV-527: Resolve handlebars to ^4.5.0

Commit:
SERV-527: Resolve handlebars to ^4.5.0

argjv committed:
TRON-133 Update account-lib dep

argjv committed:
TRON-133 Update account-lib dep

argjv committed:
Merge pull request #589 from BitGo/SERV-527-fix-audit

SERV-527: Resolve handlebars to ^4.5.0

argjv committed:
Merge pull request #589 from BitGo/SERV-527-fix-audit

SERV-527: Resolve handlebars to ^4.5.0

argjv committed:
Merge pull request #590 from BitGo/TRON-133-update-account-lib-dep

TRON-133 Update account-lib dep

argjv committed:
Merge pull request #590 from BitGo/TRON-133-update-account-lib-dep

TRON-133 Update account-lib dep

perrybitgo committed:
Fix merge conflicts of account-lib and statics version

Commit:
Merge pull request #592 from BitGo/do-555-merge

Do 555 merge

CapnMigraine committed:
Call API route for Federation lookups

The change in https://github.com/BitGo/platform/pull/1508 adds ability to perform Federation lookups through the v2.federation platform API
The SDK can now pass the params to look up the account through the BitGo Federation server instead of making the request directly to the external federation server

BLOCK-147

perrybitgo committed:
chore(release): publish modules

 - bitgo@8.5.1-rc.0
 - @bitgo/express@8.5.1-rc.0

perrybitgo committed:
chore(release): publish modules

 - bitgo@8.5.1
 - @bitgo/express@8.5.1
 - @bitgo/statics@2.4.1

argjv committed:
TRON-126 use xpub and xprv only for Tron

twtaylorbitgo committed:
Merge pull request #594 from BitGo/TRON-124-use-xpub-and-xprv

TRON-126 use xpub and xprv only for Tron

argjv committed:
Merge branch 'master' into rel/la

perrybitgo committed:
Merge pull request #595 from BitGo/rel/la

Rel/la

perrybitgo committed:
chore(release): publish modules

 - bitgo@8.5.2-rc.0
 - @bitgo/express@8.5.2-rc.0

perrybitgo committed:
chore(release): publish modules

 - bitgo@8.5.2
 - @bitgo/express@8.5.2
 - @bitgo/statics@2.4.2

Commit:
Merge pull request #596 from BitGo/rel/la

Rel/la

Commit:
BG-14814 remove v1 examples and update ts/js examples

Commit:
[ERC20-157] Add MCDai ERC20 Token

This is an upgraded version of DAI. We are adding it as a separate coin
rather than upgrading the old one because our customers need to manually
upgrade from one to the other and want to be able to swap in their own
time.

Commit:
Merge pull request #597 from BitGo/BG-14814-remove-v1-examples-and-update-ts-examples

BG-14814 remove v1 examples and update ts/js examples

marktoda committed:
Merge pull request #598 from BitGo/ERC20-157-add-mcdai

[ERC20-157] Add MCDai ERC20 Token

perrybitgo committed:
chore(release): publish modules

 - bitgo@8.6.0-rc.0
 - @bitgo/express@8.6.0-rc.0
 - @bitgo/statics@2.5.0-rc.0

Commit:
SERV-445: Update to ripple-lib@^1.4.1, add missing ripple deps

Commit:
SERV-445: Update to ripple-lib@^1.4.1, add missing ripple deps

Commit:
Update lerna and resolve handlebars to ^4.5.2

Commit:
Update lerna and resolve handlebars to ^4.5.2

Commit:
Remove node 6 and 11 from testing framework

Commit:
Remove node 6 and 11 from testing framework

Commit:
SERV-35: Add basic tests

* Ensure no dependencies are added to package.json
* Ensure all ERC20 token contract addresses are unique

twtaylorbitgo committed:
TRON-36 - trx recoveries, initial

twtaylorbitgo committed:
TRON-36 - trx recoveries, initial

Commit:
Merge pull request #601 from BitGo/SERV-445-update-ripple-lib

SERV-445: Update to ripple-lib@^1.4.1, add missing ripple deps

Commit:
Merge pull request #601 from BitGo/SERV-445-update-ripple-lib

SERV-445: Update to ripple-lib@^1.4.1, add missing ripple deps

perrybitgo committed:
Update statics version

perrybitgo committed:
chore(release): publish modules

 - bitgo@9.0.0-rc.0
 - @bitgo/express@9.0.0-rc.0

twtaylorbitgo committed:
TRON-36 - working core recover for trx

twtaylorbitgo committed:
TRON-36 - working core recover for trx

twtaylorbitgo committed:
TRON-36 - trx recovery tests

twtaylorbitgo committed:
TRON-36 - trx recovery tests

twtaylorbitgo committed:
TRON-36 - fixup from feedback

twtaylorbitgo committed:
TRON-36 - fixup from feedback

twtaylorbitgo committed:
TRON-36 - updating handlebars dep to pass npm audit

twtaylorbitgo committed:
TRON-36 - updating handlebars dep to pass npm audit

twtaylorbitgo committed:
TRON-36 - prettier run

twtaylorbitgo committed:
TRON-36 - prettier run

twtaylorbitgo committed:
Merge pull request #603 from BitGo/TRON-36-trx-recoveries

TRON-36 - trx recoveries, initial

twtaylorbitgo committed:
Merge pull request #603 from BitGo/TRON-36-trx-recoveries

TRON-36 - trx recoveries, initial

twtaylorbitgo committed:
Merge pull request #602 from BitGo/SERV-35-statics-tests

SERV-35: Add basic tests

perrybitgo committed:
chore(release): publish modules

 - bitgo@9.0.0-rc.1
 - @bitgo/express@9.0.0-rc.1

perrybitgo committed:
chore(release): publish modules

 - bitgo@9.0.0
 - @bitgo/express@9.0.0
 - @bitgo/statics@2.5.0

perrybitgo committed:
Update SDK to bitgo@^9.0.0

twtaylorbitgo committed:
TRON-36 - fixing tron recoveries for WRW

twtaylorbitgo committed:
TRON-36 - fixing tron recoveries for WRW

twtaylorbitgo committed:
Merge pull request #605 from BitGo/TRON-36-recoveries-typings

TRON-36 - tron WRW fixup

twtaylorbitgo committed:
Merge pull request #605 from BitGo/TRON-36-recoveries-typings

TRON-36 - tron WRW fixup

perrybitgo committed:
Update statics version

perrybitgo committed:
chore(release): publish modules

 - bitgo@9.0.1-rc.0
 - @bitgo/express@9.0.1-rc.0

Commit:
BG-14806 reject hop param for erc-20 token build

Commit:
[BLOCK-180] Add ERC20 Tokens

BitNational (BNL) -- 0xa717d0f45652fb430fd84d3d1b6b02e4510102ea
Paxos Gold (PAXG) -- 0x45804880De22913dAFE09f4980848ECE6EcbAf78
Zipmex Token (ZMT) -- 0x79adc3ba33625e3e801e2ba9cba83e43c555c797
Rupiah Token (IDRT) -- 0x998FFE1E43fAcffb941dc337dD0468d52bA5b48A
Caspian Token (CSP) -- 0xa6446d655a0c34bc4f05042ee88170d056cbaf45
MenaPay (MPAY) -- 0x3810a4ddf41e586fa0dba1463a7951b748cecfca
Compound WBTC (CWBTC) -- 0xC11b1268C1A384e55C48c2391d8d480264A3A7F4

Commit:
[BLOCK-180] Add ERC20 Tokens

BitNational (BNL) -- 0xa717d0f45652fb430fd84d3d1b6b02e4510102ea
Paxos Gold (PAXG) -- 0x45804880De22913dAFE09f4980848ECE6EcbAf78
Zipmex Token (ZMT) -- 0x79adc3ba33625e3e801e2ba9cba83e43c555c797
Rupiah Token (IDRT) -- 0x998FFE1E43fAcffb941dc337dD0468d52bA5b48A
Caspian Token (CSP) -- 0xa6446d655a0c34bc4f05042ee88170d056cbaf45
MenaPay (MPAY) -- 0x3810a4ddf41e586fa0dba1463a7951b748cecfca
Compound WBTC (CWBTC) -- 0xC11b1268C1A384e55C48c2391d8d480264A3A7F4

marktoda committed:
Merge pull request #609 from BitGo/BLOCK-180-add-ERC20-Tokens

[BLOCK-180] Add ERC20 Tokens

marktoda committed:
Merge pull request #609 from BitGo/BLOCK-180-add-ERC20-Tokens

[BLOCK-180] Add ERC20 Tokens

champbronc2 reviewed (approved):
Since this is already live on production, I assume we are good here?

If so, consider this approved

perrybitgo committed:
chore(release): publish modules

 - bitgo@9.0.1-rc.1
 - @bitgo/express@9.0.1-rc.1
 - @bitgo/statics@2.6.0-rc.0

dannydeezy committed:
Merge pull request #607 from BitGo/BG-14806-verbose-error-on-erc-hop-fail

BG-14806 reject hop param for erc-20 token build

CapnMigraine committed:
Merge pull request #593 from BitGo/BLOCK-147.call-api-stellar-federation-lookups

BLOCK-147: Call API route for Federation lookups

perrybitgo committed:
chore(release): publish modules

 - bitgo@9.0.1
 - @bitgo/express@9.0.1
 - @bitgo/statics@2.6.0

perrybitgo committed:
chore(release): publish modules

 - bitgo@9.1.0-rc.0
 - @bitgo/express@9.1.0-rc.0
 - @bitgo/statics@2.7.0-rc.0

cooncesean committed:
Clean Up DEVELOPERS.md

- Fix formatting in DEVELOPERS.md, "Things To Avoid" section.
- Clean up grammar/readability in "Things To Avoid" section.

Commit:
Merge branch 'master' into merge-manchester

Commit:
Merge pull request #612 from BitGo/merge-manchester

Merge manchester

Commit:
SERV-579 filter out duplicate addresses in cross-chain

cooncesean committed:
Add GitHub Issue Templates

Add two GitHub issues templates to this project (https://help.github.com/en/github/building-a-strong-community/configuring-issue-templates-for-your-repository):

1. Bugs: Add a standardized template for contributors to report bugs to
the project maintainers.
2. Feature Requests: Add a template that allows contributors to request
features.

....

Motivation: There is no standardized way for people to report their
issues to this project. When bugs are reported, they tend to omit
critical details such as node version, os, etc. which are extremely
helpful in diagnosing the issue.

cooncesean committed:
Remove extraneous issue template markdown

Commit:
Properly deserialize EOS staking transactions

JIRA: ETSTK-11

This commit adds functionality to deserialize EOS staking
transactions. EOS staking transactions consist of two actions,
delegatebw and voteproducer. The delegatebw action specifies the
amount of EOS to be staked and the account to stake it for. In our
case the staker and the receiver of the stake will always be the
same. The voteproducer action specifies the proxy or producers that
the staked funds will vote for.

cooncesean committed:
Add a link to our documentation.

cooncesean committed:
Merge pull request #611 from BitGo/patch-developers.md

SERV-581 Clean Up `DEVELOPERS.md`

Commit:
BLOCK-200 fix eos address validation

perrybitgo committed:
chore(release): publish modules

 - bitgo@9.1.0
 - @bitgo/express@9.1.0
 - @bitgo/statics@2.7.0

perrybitgo committed:
Update SDK to bitgo@^9.1.0

perrybitgo committed:
Update release version

Commit:
Merge branch 'master' into merge-naples

Commit:
Merge pull request #617 from BitGo/merge-naples

Merge naples

cooncesean committed:
Merge pull request #614 from BitGo/add-issue-tracker-template

SERV-582 Add issue tracker template

Commit:
BG-14844 Fix accelerateTx circular json bug

For some reason a wallet object is getting added to the params before we do submitTransaction, and this is causing a failure as it shouldn't be there. This just deletes that wallet param and fixes the issue.

dannydeezy committed:
Merge pull request #618 from BitGo/BG-14844-fix-cpfp-circular-json-error

BG-14844 Fix accelerateTx circular json bug

james8nguyen committed:
BLOCK-226 Updated bitgo-utxo-lib version to 1.7.0-rc / 1.7.0 which supports new Blossom network params.

Commit:
CT-489: Check for wrapped segwit unspents in express v1 integration test

Commit:
CT-489: Check for wrapped segwit unspents in express v1 integration test

Commit:
CT-489: Merge pull request #625 from BitGo/CT-489-fix-broken-express-v1-test

CT-489: Check for wrapped segwit unspents in express v1 integration test

Commit:
Merge pull request #625 from BitGo/CT-489-fix-broken-express-v1-test

CT-489: Check for wrapped segwit unspents in express v1 integration test

james8nguyen committed:
BLOCK-226 Merge pull request #624 from BitGo/BLOCK-226-update-utxo-version

BLOCK-226 Update bitgo-utxo-lib to use new Zcash params.

james8nguyen committed:
BLOCK-226 Update bitgo-utxo-lib to use new Zcash params.

james8nguyen committed:
BLOCK-226 Merge pull request #626 from BitGo/BLOCK-226-update-utxo

BLOCK-226 Update bitgo-utxo-lib to use new Zcash params.

james8nguyen committed:
chore(release): publish modules

 - bitgo@9.2.0-rc.0
 - @bitgo/express@9.2.0-rc.0

Commit:
Merge pull request #619 from BitGo/ETSTK-11-eos-staking-deserialization

Properly deserialize EOS staking transactions

perrybitgo committed:
chore(release): publish modules

 - bitgo@9.2.0
 - @bitgo/express@9.2.0

perrybitgo committed:
Update SDK to bitgo@^9.2.0

perrybitgo committed:
Update release version

Commit:
Merge pull request #627 from BitGo/rel/naples

Rel/naples

Commit:
[BLOCK-237] Add ERC20 Tokens

Wrapped XRP (WXRP) -- 0xda7e5a3841550a5ba271dcc76a885af902142dfc
Saga (SGA) -- 0xed0849bf46cfb9845a2d900a0a4e593f2dd3673c
Orchid (OXT) -- 0x4575f41308EC1483f3d399aa9a2826d74Da13Deb
Binance USD (BUSD) -- 0x4Fabb145d64652a948d72533023f6E7A623C7C53

marktoda committed:
Merge pull request #629 from BitGo/BLOCK-237-add-ERC20-Tokens

[BLOCK-237] Add ERC20 Tokens

dannydeezy committed:
Merge pull request #613 from BitGo/SERV-579-filter-out-dup-addr-in-cross-chain

SERV-579 filter out duplicate addresses in cross-chain

dannydeezy committed:
Merge pull request #615 from BitGo/BLOCK-200-fix-eos-addr-validation

BLOCK-200 fix eos address validation

Commit:
fix(SERV-593): Correctly handle undefined boolean config items

Boolean flags for bitgo-express (eg, `disableSSL`, `disableProxy`, etc)
currently have no effect.

The issue is that the environment config, which is highest in
precedence, casts undefined env vars to booleans if the config field is
of boolean type. This means the lower precedence configs are not
considered (args and default configs). Therefore, all command line flags
which set booleans are not able to be set, and instead must be set via
environment instead.

fixes #599

Commit:
Merge pull request #630 from BitGo/SERV-593-express-params

fix(SERV-593): Correctly handle undefined boolean config items

Commit:
fix(SERV-597): Ensure `Error.captureStackTrace` is defined before call

The `Error.captureStackTrace` function is only available in V8, so error
construction fails in other JS runtimes. This makes it difficult to
determine the root cause of errors. Instead, we should only call this if
it is defined.

Commit:
Merge pull request #631 from BitGo/SERV-597-fix-error-constructor

fix(SERV-597): Ensure `Error.captureStackTrace` is defined before call

Commit:
Return algo keyreg data from explainTransaction

This commit returns type, senderAddress, voteKeyBase64, and
voteLastBlock from explainTransaction. The OVC needs this data to verify
what it is signing.

JIRA: AIR-6

Commit:
Merge pull request #632 from BitGo/AIR-6-algo-keyreg-deserialization

Return algo keyreg data from explainTransaction

Commit:
[BLOCK-236] Remove unsupported ERC20 tokens

There are several ERC20 tokens in statics that are not supported by the
rest of the BitGo platform. In an effort to integrate statics with other
components, we need to remove those extraneous tokens, otherwise we will
accidentally support them.

Tokens removed:
ABT, AGI, AUTO, BLZ, BNB, BST, BTM, CENNZ, CMT, CND, CS, DCN, DEW, DRGN,
DROP, DTR, ENG, ERC, FSN, GNX, GVT, HPB, ICN, ICX, IOST, KCS, MAN, MANA,
NCASH, NULS, OST, PAYX, PLR, POE, PPP, PRL, R, REQ, RHOC, RUFF, SAN,
SMT, SRN, SUB, THETA, TMS, VEE, VEN, VERI, XIN

marktoda committed:
Merge pull request #633 from BitGo/BLOCK-236-update-statics-to-match-platform

[BLOCK-236] Remove unsupported ERC20 tokens
